src/thunderbots/software/ai/hl/stp/play/halt_play.cpp

```diff
@@ -0,0 +1,49 @@
+#include "ai/hl/stp/play/halt_play.h"
+
+#include "ai/hl/stp/play/play_factory.h"
+#include "ai/hl/stp/tactic/stop_tactic.h"
+
+const std::string HaltPlay::name = "Halt Play";
+
+std::string HaltPlay::getName() const
+{
+    return HaltPlay::name;
+}
+
+bool HaltPlay::isApplicable(const World &world) const
+{
+    return world.gameState().isHalted();
+}
+
+bool HaltPlay::invariantHolds(const World &world) const
+{
+    return world.gameState().isHalted();
+}
+
+void HaltPlay::getNextTactics(TacticCoroutine::push_type &yield)
```

Comment:
As a note for the future, we should make this take the number of robots to stop in the constructor.

Reply:
why? The plays don't have much control over their construction, it's done automatically during play selection